WooCommerce/Classes/Authentication/AuthenticationManager.swift

```diff
@@ -264,18 +264,16 @@ extension AuthenticationManager: WordPressAuthenticatorDelegate {
             return
         }
 
-        /// Jetpack is required. Present an error if we don't detect a valid installation for a self-hosted site.
-        if let site = currentSelfHostedSite,
-            site.url == wpcomLogin.siteURL, !site.hasValidJetpack {
-            let viewModel = JetpackErrorViewModel(siteURL: wpcomLogin.siteURL)
-            let installJetpackUI = ULErrorViewController(viewModel: viewModel)
-            return navigationController.show(installJetpackUI, sender: nil)
-        }
-
         let matcher = ULAccountMatcher()
         matcher.refreshStoredSites()
 
         guard matcher.match(originalURL: wpcomLogin.siteURL) else {
+
+            /// Jetpack is required. Present an error if we don't detect a valid installation for a self-hosted site.
+            if checkJetpackErrorForSelfHostedSite(url: wpcomLogin.siteURL) {
+                presentJetpackError(for: wpcomLogin.siteURL, with: credentials, in: navigationController, onDismiss: onDismiss)
```

Comment:
I think we should `return` after this line to avoid executing the following lines, which will end up showing `WrongAccountErrorViewModel` after showing no Jetpack error.